lib/iris/aux_factory.py

```diff
@@ -784,20 +758,51 @@ def dependencies(self):
                     depth_c=self.depth_c, nsigma=self.nsigma, zlev=self.zlev)
 
     def _derive(self, sigma, eta, depth, depth_c,
-                zlev, shape, nsigma_slice):
+                zlev, nsigma, coord_dims_func):
+        # Calculate the index of the 'z' dimension in the inputs.
+        # Get the cube dimension...
+        i_levels_cubedim, = coord_dims_func(self.dependencies['zlev'])
+        # ...subtract any previous dimensions not used in the calculation.
+        derived_cubedims = self.derived_dims(coord_dims_func)
+        i_levels_dim = i_levels_cubedim - sum(
+            i_dim not in derived_cubedims
+            for i_dim in range(i_levels_cubedim))
+
+        # Calculate the result shape as a combination of all the inputs.
+        allshapes = np.array(
+            [el.shape
+             for el in (sigma, eta, depth, depth_c, zlev)
+             if el.ndim])
+        result_shape = list(np.max(allshapes, axis=0))
```

Comment:
@pp-mo This only works if **all** the `el`ements have the **same** number of `shape` dimensions ... which must be the case, right? ... given the output from `_remap` and `_remap_with_bounds`, which aligns the dimensionality of everything or injects 0-d scalars for missing coordinates.

Just looking for re-assurance (and convince me otherwise) but should we ensure that `allshapes` has equal length elements before doing the `np.max` ?